rfc-120-csv-preview.md

```diff
@@ -0,0 +1,134 @@
+# CSV Preview
+
+[govuk-content-schemas]: https://github.com/alphagov/govuk-content-schemas
+[government-prefix]: https://trello.com/c/74KTVXBs/52-whitehall-relies-on-prefix-government-and-world-routes
+[router]: https://github.com/alphagov/router#backend-handler
+
+## Summary
+
+Move the CSV Preview feature out of Whitehall and into Government Frontend, so that it can be used for documents published by other apps, like Content Publisher. The new implementation will work by rendering a content item in a new `attachment_preview` format, which represents an attachment to preview.
+
+## Problem
+
+A user viewing a Whitehall document on GOV.UK can click a "View Online" link for any of its CSV file attachments. This takes the user to a new page, currently rendered by Whitehall. The page contains:
+
+   - A 'header' section with details of the parent document (and navigation links)
+   - A 'body' section with a preview of the content of the CSV attachment
+
+This CSV Preview feature only works for CSV file attachments uploaded through Whitehall. We need it to work for CSV attachments published through Content Publisher, which is superseding Whitehall.
+
+## Proposal
+
+### Step 1
+
+Create a new `attachment_preview` format in [govuk-content-schemas][] to represent the preview page for an attachment. A publishing app will publish content items in this format for each CSV attachment to preview.
```

Comment:
Do we want to consider specifying what we recommend using for update_type? It's probably something of an implementation detail just one of the questions that comes to my mind when thinking about it as a content item, similarly for public_updated_at timestamp. Sorry it this is a bit of tangential comment, maybe can just copy HTML attachment approaches.

Reply:
Tricky. I just realised I'd inherited from `default_format`, but we actually want `publishing_api_base`, since we'll (at least) want `access_limited` stuff.

So, while I've fixed that, I do think `update_type`, `change_note`, and `public_updated_at` are more of an implementation detail.